maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java

```diff
@@ -150,6 +152,8 @@ public DefaultRepositorySystemSession newRepositorySession( MavenExecutionReques
         configProps.put( ConfigurationProperties.USER_AGENT, getUserAgent() );
         configProps.put( ConfigurationProperties.INTERACTIVE, request.isInteractiveMode() );
         configProps.put( "maven.startTime", request.getStartTime() );
+        // First add properties populated from settings.xml
+        configProps.putAll( getConfigFromRequestProfiles( request ) );
         // Resolver's ConfigUtils solely rely on config properties, that is why we need to add both here as well.
```

Comment:
I think this order also complies with preferred inheritance at point 7: https://cwiki.apache.org/confluence/display/MAVEN/Commandline+inheritance

Reply:
so order is correct